factory-girl/src/main/scala/skinny/test/FactoryGirl.scala

```diff
@@ -51,8 +53,22 @@ case class FactoryGirl[Id, Entity](mapper: CRUDFeatureWithId[Id, Entity], name:
    * @return attributes in conf file
    */
   def loadedAttributes(): Map[SQLSyntax, Any] = {
-    // TODO directory scan and work with factories/*.conf
-    val config = ConfigFactory.parseResources(getClass.getClassLoader, "factories.conf").resolve().getConfig(factoryName.name)
+    val rootResolve = ConfigFactory.parseResources(getClass.getClassLoader, "factories.conf").resolve()
+    val confDir = "factories"
+
+    val resolves = getClass.getClassLoader
+      .getResources(confDir)
+      .asScala.flatMap { uri =>
+        new File(uri.getFile).listFiles
+          .withFilter { f => f.isFile && f.getName.endsWith(".conf") }
+          .map { f => s"${confDir}/${f.getName}" }
+      }.map(ConfigFactory.parseResources(getClass.getClassLoader, _).resolve()).toSeq
+
+    val config = (rootResolve +: resolves)
+      .collectFirst({
+        case c if Try { c.getConfig(factoryName.name) }.isSuccess => c.getConfig(factoryName.name)
```

Comment:
When we have lots of conf files under `factories` directory, this method can be a bottleneck because it will be called each time we use FactoryGirl.create. How about caching the result?